### PR TITLE
bpo-41429: Let fnmatch.filter accept a tuple of patterns

### PR DESCRIPTION
**Pull request by andresdelfino**
https://bugs.python.org/issue41429

### CODE REVIEW

File: Doc/library/fnmatch.rst [hunk 1]
```diff
@@ -75,7 +75,8 @@ patterns.
 
 .. function:: filter(names, pattern)
 
-   Return the subset of the list of *names* that match *pattern*. It is the same as
+   Return the subset of the list of *names* that match *pattern*.  *pattern*
+   can be a tuple of patterns.  It is the same as
```
Comment (ZackerySpytz) [inline Doc/library/fnmatch.rst]: Please add a `versionchanged` directive for the new changes in behavior.

File: Lib/test/test_fnmatch.py [hunk 2]
```diff
@@ -152,6 +152,12 @@ def test_filter(self):
         self.assertEqual(filter([b'Python', b'Ruby', b'Perl', b'Tcl'], b'P*'),
                          [b'Python', b'Perl'])
 
+    def test_filter_multipattern(self):
+        self.assertEqual(filter(['Python', 'Ruby', 'Perl', 'Tcl'], ('P*', 'T*')),
+                         ['Python', 'Perl', 'Tcl'])
+        self.assertEqual(filter([b'Python', b'Ruby', b'Perl', b'Tcl'], (b'P*', b'T*')),
```
Comment (ZackerySpytz) [inline Lib/test/test_fnmatch.py]: The PEP 8 style guide states that lines should be limited to 79 characters.

File: Doc/library/fnmatch.rst (hunk 1 above)
Comment (andresdelfino) [reply Doc/library/fnmatch.rst]: Addressed your comments on Dec 13, 2020 and forgot to tell you :sweat_smile:

File: Lib/test/test_fnmatch.py (hunk 2 above)
Comment (andresdelfino) [reply Lib/test/test_fnmatch.py]: Addressed your comments on Dec 13, 2020 and forgot to tell you :sweat_smile:

**Comment by andresdelfino** [comment]
@ZackerySpytz I have addressed all your comments  :)

File: Lib/fnmatch.py [hunk 3]
```diff
@@ -54,17 +54,24 @@ def _compile_pattern(pat):
 def filter(names, pat):
     """Construct a list from those elements of the iterable NAMES that match PAT."""
     result = []
-    pat = os.path.normcase(pat)
-    match = _compile_pattern(pat)
+    pats = pat if isinstance(pat, tuple) else (pat,)
```
Comment (phdru) [inline Lib/fnmatch.py:57]: My advice is to allow any iterable, not just tuple:

    pats = (pat,) if isinstance(pat, str) else pat  # Assume ``pat`` is an iterable

**Review by phdru — CHANGES_REQUESTED** [review]

**Pull request closed without merging**